core/serialization/deserializer/montage-deserializer.js

```diff
@@ -76,9 +95,39 @@ var MontageDeserializer = exports.MontageDeserializer = Montage.specialize({
 
     preloadModules: {
         value: function () {
-            var serialization = JSON.parse(this._serializationString);
+            var serialization = JSON.parse(this._serializationString),
```

Comment:
JSON.parse(this._serializationString) is also executed in deserialize(), can't that be done only once?

Reply:
Something strange is happening. If we just do the parse once in the init instead of doing it at the beginning of preloadModules() and deserialize(), a bunch of tests fail with "Converting circular structure to JSON" errors. I think it's because in the case where the deserializer is passed an object instead of a string, the result of JSON.parse-ing the JSON.stringify-ed object is different from the original object. So for now when the deserializer is initialized with an object, it has to stringify() it so it can then parse() it and avoid these errors. This shouldn't be the case, but I couldn't figure out the solution yet, we should shelve this change for another PR.